lib/matplotlib/backend_managers.py

```diff
@@ -245,6 +283,8 @@ def add_tool(self, name, tool, *args, **kwargs):
             else:
                 self._toggled.setdefault(tool_obj.radio_group, None)
 
+        if self.figure:
+            tool_obj.set_figure(self.figure)
```

Comment:
Sorry, I said that awkwardly, I meant to say "do we really need the if?", i.e. keep the second line in there.  If ``self.figure == None``, then the call to tool_obj.set_figure(self.figure) just does a no-op...